Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/aspect/EventAspectAdvice.java

```diff
@@ -47,123 +48,124 @@
+     * END_* event.
+     *
+     * If the exception is an instance of ErrorEventException, then the event will be logged as a failure, but the
+     * exception will be suppressed. This will allow for services such as Document Query in which the response must
+     * contain a RegistryErrorList body element in the SOAP response and should not produce a SOAP fault. In such
+     * situations, the ErrorEventException should return an override object to be used as the result of the method
+     * invocation. This will allow the service to both log the exception raised, log the failed processing event, suppress
+     * logging the corresponding END_* event, and satisfy the requirements of the given service spec by returning a
+     * RegistryErrorList containing the errors that occurred.
+     *
+     * If the exception is any other type other than ErrorEventException, the event will be logged as a failure, with
+     * no corresponding END_* event and the exception will be re-thrown.
+     *
+     * @param joinPoint
+     * @param delegate
+     * @param beforeBuilder
+     * @param afterBuilder
+     * @param serviceType
+     * @param version
+     * @return
+     * @throws Throwable
      */
-    @AfterThrowing(pointcut = "@annotation(gov.hhs.fha.nhinc.aspect.InboundMessageEvent) ||"
-        + "@annotation(gov.hhs.fha.nhinc.aspect.InboundProcessingEvent) ||"
-        + "@annotation(gov.hhs.fha.nhinc.aspect.AdapterDelegationEvent) || "
-        + "@annotation(gov.hhs.fha.nhinc.aspect.OutboundMessageEvent) || "
-        + "@annotation(gov.hhs.fha.nhinc.aspect.OutboundProcessingEvent) || "
-        + "@annotation(gov.hhs.fha.nhinc.aspect.NwhinInvocationEvent)", throwing = "e")
-    public void failEvent(JoinPoint joinPoint, Throwable e) {
-        if (eventRecorder != null && eventRecorder.isRecordEventEnabled()) {
-            MethodSignature sig = (MethodSignature) joinPoint.getSignature();
-            Method method = sig.getMethod();
-
-            // Check if the method is annotated with @LogFailures. We dont want to log the event twice.
-            if (method.getAnnotation(LogFailures.class) != null) {
-                Logger log = LoggerFactory.getLogger(sig.getDeclaringType()); //NOSONAR
-                log.warn("Method {} has multiple logging aspects. Will only process LogFailures annotation... ", method.getName());
-            } else {
-                logFailure(joinPoint, e);
-            }
+    public Object logEvents(ProceedingJoinPoint joinPoint, EventAdviceDelegate delegate, Class<? extends EventDescriptionBuilder> beforeBuilder,
+        Class<? extends EventDescriptionBuilder> afterBuilder, String serviceType, String version) throws Throwable {
+        delegate.begin(joinPoint.getArgs(), serviceType, version, beforeBuilder);
+
+        Object value = null;
+        boolean hasFailure = false;
+        try {
+            value = joinPoint.proceed();
+        } catch (ErrorEventException e) {
```

Comment:
Question: I don't see where we throw this ErrorEventException to catch it.

Reply:
This is going to be thrown in the next story, which we will be discussing at 3 PM today.